src/category_theory/sites/sieves.lean

```diff
@@ -438,9 +444,24 @@ begin
     exact ⟨_, _, _, presieve.pullback_arrows.mk _ _ hk, pullback.lift_snd _ _ comm⟩ },
 end
 
+/--
+If `R` is a sieve, then the `category_theory.presieve.functor_pullback` of `R` is actually a sieve.
+-/
+def functor_pullback {X : C} (R : sieve (F.obj X)) : sieve X :=
+{ arrows := presieve.functor_pullback F R,
+  downward_closed' := λ _ _ f hf g,
+  begin
+    unfold presieve.functor_pullback,
+    rw F.map_comp,
+    exact R.downward_closed hf (F.map g),
+  end }
+
+@[simp] lemma functor_pullback_id {X : C} (R : sieve X) : R.functor_pullback (𝟭 _) = R :=
```

Comment:
Maybe make this lemma for presieves instead? Together with `sieve.ext`, this gives the lemma for sieves for free.

Reply:
It is defeq for the presieves case, I'm not sure if that needs a lemma.

Reply:
But how can we apply this lemma if we only know that `R` is a presieve?

Reply:
Oh I have misunderstood you: I thought you meant this lemma would be defeq to the one for presieves. Anyway, I think it's good to let simp know about `functor_pullback_id` for presieves, even if it's just `rfl`.